### PR TITLE
[MM-16677] New unsupported browser page

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
This PR adds a new unsupported browser page to the app that will appear when Internet Explorer or an outdated version of Safari is used to access Mattermost. This appears as we are dropping support for Internet Explorer in v5.16

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-17569
https://mattermost.atlassian.net/browse/MM-17571
https://mattermost.atlassian.net/browse/MM-17572
https://mattermost.atlassian.net/browse/MM-17573

### CODE REVIEW

**Comment by devinbinnie** [comment]
https://github.com/mattermost/mattermost-webapp/pull/3416 will need to be merged first as it contains the images used by the page.

File: web/web.go [hunk 1]
```diff
@@ -43,15 +43,16 @@ func New(config configservice.ConfigService, globalOptions app.AppOptionCreator,
 // Due to the complexities of UA detection and the ramifications of a misdetection
 // only older Safari and IE browsers throw incompatibility errors.
 // Map should be of minimum required browser version.
+// -1 means that the browser is not supported in any version.
 var browserMinimumSupported = map[string]int{
-	"BrowserIE":     11,
+	"BrowserIE":     12,
```
Comment (hmhealey) [inline web/web.go]: Should this be -1?

**Comment by andrewbrown00** [comment]
@devinbinnie can you include screen shots or a link to your server for me to review?

File: web/web.go (hunk 1 above)
Comment (devinbinnie) [reply web/web.go]: That's originally what I thought, which is why I added that functionality.
But turns out Edge is interpreted by `uasurfer` as IE12, so we have to keep it this way.

**Comment by devinbinnie** [comment]
> @devinbinnie can you include screen shots or a link to your server for me to review?

@andrewbrown00 I can as soon as https://github.com/mattermost/mattermost-webapp/pull/3416 is approved, as we need the images in the webapp otherwise the page will not display properly.

**Comment by andrewbrown00** [comment]
Thanks!

File: web/web.go (hunk 1 above)
Comment (hmhealey) [reply web/web.go]: Despite Microsoft's best efforts to Edge from IE, they can never get away from it fully 😛

**Comment by devinbinnie** [comment]
@andrewbrown00 Server is up and ready to test!

**Comment by andrewbrown00** [comment]
@devinbinnie IE11 is working as expected, but when I hit the SpinWick link for legacy browsers it takes me to a log in screen, instead of https://invis.io/X9SXUD8KUAB#/373424124_macOS_-_Unsupported_Browser_Version

**Test on macOS with**
- Chrome 55
- Firefox 52

**Comment by devinbinnie** [comment]
@andrewbrown00 So there's a comment in the code that reads: "Due to the complexities of UA detection and the ramifications of a misdetection, only older Safari and IE browsers throw incompatibility errors."

I'm not sure if this is still true, but that's how the previous unsupported browser page functioned, so I decided to keep that true for the current one, since we're still using the same library to read the User Agent string.

The comment was made by @jwilander about a year ago, perhaps he could shed some further light on this?

**Comment by andrewbrown00** [comment]
@devinbinnie thanks for bringing this to my attention. Let's pull in PM and see how they want to proceed? In the meantime, what do you think the mana cost would be to try to improve the UA detection to handle Chrome and Firefox?

@esethna take a look at the last few comments regarding **UA detection** and let us know how you'd like to proceed:

1. Leave the behavior as is, which means we'll let users open MM in unsupported browser versions without informing them.
2. Try to enhance the UA detection to include Chrome and Firefox unless we think this is going to be more of a corner case and not worth the mana?
   -  I'm 3/5 they're used more often, but can we pull the data on browser usage to confirm.
3. Any other thoughts...

**Comment by devinbinnie** [comment]
@andrewbrown00 Well to add the settings, it would be 0 mana, it's a really easy add. But given that comment we might need to test on a few different browsers. Some lesser known browsers (for example, Konqueror) might not be detected properly.

**Comment by devinbinnie** [comment]
Another note about Chrome and Firefox: Both browsers have auto update mechanisms that should keep them up to date, so it's rare that people will be using out of date versions of those browsers, unless their OS only supports up to a certain version.

**Comment by andrewbrown00** [comment]
@devinbinnie that's great on the mana cost. I'm 4/5 we would only need to test on the browsers we support unless there's a technical reason to test on other browsers?
- Do we have any detection for unsupported browsers like Konqueror or Opera today? 

I agree about the auto updates, I'm wondering if it's worth having the detection for the cases that don't have auto updates enabled for whatever reason. That way we avoid users running into bugs or less optimal experiences on a browser version we don't support.

Can you follow up with @jwilander to see if there's an explicit reason we didn't invest in adding detection that may still stand today?

**Comment by devinbinnie** [comment]
@andrewbrown00 We don't currently have anything for either of those browsers, but I know Opera is supported by `uasurfer`, Konqueror however isn't mentioned in the docs at least.

**Comment by devinbinnie** [comment]
@andrewbrown00 @esethna Turns out it was @hmhealey who owned this way back when, and after speaking with him it turns out it was added to deal with misdetections that were reported after the fact. One of them was even our desktop app. See this PR comment: https://github.com/mattermost/mattermost-server/pull/7427#issuecomment-329930008
and this ticket: https://mattermost.atlassian.net/browse/MM-7319

It seems to me that we shouldn't be restricting people from using the app with slightly older versions of modern browsers, we should only restrict if we know that the app will not function properly (ie. older versions of Safari, Internet Explorer). Most versions of Chrome/Firefox should work with the app correctly. The supported version for Chrome/Firefox exists (I believe) so that we don't have to fix bugs that happened on a specific older Chrome/Firefox version.

Let me know your thoughts.

**Comment by andrewbrown00** [comment]
@devinbinnie everything is looking good for the IE11 unsupported work. 
image

**Item to resolve**
The `Make default` action for on the Edge card isn't working for me, it doesn't open the app settings after clicking the link.
image

**Comment by devinbinnie** [comment]
@andrewbrown00 Works fine for me, what version of Windows 10 are you running?
Can anyone else reproduce? @hmhealey?

**Comment by andrewbrown00** [comment]
@devinbinnie I just closed IE11 and retested and it worked fine. Maybe it was in a dirty state? I tested it a few more times to try and repro the issue but I couldn't.

All good ✅

**Pull request merged**